lib/printer.js

```diff
@@ -579,9 +583,9 @@ function genericPrintNoParens(path, options, print) {
         var oneLine = (isTypeAnnotation && len === 1) || len === 0;
         var parts = [oneLine ? "{" : "{\n"];
 
+        var i = 0;
         fields.forEach(function(field) {
             path.map(function(childPath) {
-                var i = childPath.getName();
```

Comment:
because we are iterating over three different fields for `ObjectTypeAnnotation` this counter was wrong and reset for every individual field (`ObjectTypeAnnotation` is a really nasty type). I made it so that it counts the fields in the order they are printed. This fixes additional newlines that were accidentally printed in the object.